polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -624,6 +624,10 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *Expr) {
       assert(Nt.getAddressSpace() == MT.getMemorySpaceAsInt() &&
              "val does not have the same memory space as nt");
       Val = Builder.create<polygeist::Memref2PointerOp>(Loc, Nt, Val);
+    } else if (auto PT = dyn_cast<LLVM::LLVMPointerType>(Val.getType())) {
+      if (!UseOpaquePointers)
+        if (auto ST = dyn_cast<polygeist::StructType>(PT.getElementType()))
+          Val = Builder.create<LLVM::BitcastOp>(Loc, LLVMTy, Val);
```

Comment:
Why is this needed? Shouldn't we be using `memref` to host `polygeist.struct` now that `llvm.struct` are no longer used?

Reply:
In this PR, we are not modifying the existing `!llvm.ptr<!polygeist.struct>` to `memref<?x!polygeist.struct>`.